### PR TITLE
Adding support for references to another peripheral

### PR DESCRIPTION
**Pull request by idubrov**
Handling "[PERIPHERAL].[REGISTER].[FIELD].[ENUMERATEDVALUES]" format
for the "derivedFrom" attribute in <enumeratedValues>.

(used https://www.keil.com/pack/doc/CMSIS/SVD/html/elem_registers.html#elem_enumeratedValues as a reference).

### CODE REVIEW

File: src/generate.rs [hunk 1]
```diff
@@ -767,7 +772,19 @@ pub fn fields(
...
+                    if let (Some(ref peripheral), Some(ref register)) = (base.peripheral, base.register) {
+                        let mod1_ = peripheral.to_sanitized_snake_case();
+                        let mod2_ = register.to_sanitized_snake_case();
+                        let mod1_ = Ident::new(&*mod1_);
+                        let mod2_ = Ident::new(&*mod2_);
+
+                        mod_items.push(
+                            quote! {
+                                #[doc = #description]
+                                pub type #pc_r = super::super::#mod1_::#mod2_::#base_pc_r;
```
Comment (japaric) [inline src/generate.rs]: nit: I think the super::super is not necessary as all the peripheral modules are in the root of the crate.

File: src/generate.rs [hunk 2]
```diff
@@ -1058,7 +1076,24 @@ pub fn fields(
...
+                                let mod1_ = peripheral.to_sanitized_snake_case();
+                                let mod2_ = register.to_sanitized_snake_case();
+                                let mod1_ = Ident::new(&*mod1_);
+                                let mod2_ = Ident::new(&*mod2_);
+
+                                mod_items.push(
+                                    quote! {
+                                        #[doc = #pc_w_doc]
+                                        pub type #pc_w =
+                                            super::super::#mod1_::#mod2_::#base_pc_w;
```
Comment (japaric) [inline src/generate.rs]: nit: same comment about super::super as above

File: src/generate.rs [hunk 3]
```diff
@@ -1058,7 +1076,24 @@ pub fn fields(
...
+                                    },
+                                );
+
+                                quote! {
+                                    super::super::#mod1_::#mod2_::#base_pc_w
```
Comment (japaric) [inline src/generate.rs]: nit: same comment about super::super as above

**Comment by japaric** [comment]
Thanks for the PR @idubrov. This looks good to me. I left a comment about the use of super::super which I think is not necessary.

**Comment by idubrov** [comment]
Replaced super::super:: with :: (I'm not sure I had to do it -- my knowledge of Rust is currently at the level of "copy-paste monkey"). Also, renamed mod1/mod2 to _slightly_ more descriptive pmod/rmod.

**Comment by japaric** [comment]
Thanks, @idubrov!

@homunkulus r+

P.S. Please next time push a new commit when you do a change instead of rebasing the last one. That makes reviewing easier.

**Comment by homunkulus** [comment]
:pushpin: Commit 968fbc5 has been approved by `japaric`

**Comment by homunkulus** [comment]
:hourglass: Testing commit 968fbc5 with merge 968fbc5...

**Comment by homunkulus** [comment]
:sunny: Test successful - [status-appveyor](https://ci.appveyor.com/project/japaric/svd2rust/build/1.0.75), [status-travis](https://travis-ci.org/japaric/svd2rust/builds/235728776?utm_source=github_status&utm_medium=notification)
Approved by: japaric
Pushing 968fbc5da03974bc9b9956dec4509fc4b03d784a to master...

**Pull request merged**